File: backend/app/schemas/cliente.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr


class ClienteBase(BaseModel):
    nome: str
    email: EmailStr
    telefone: str | None = None


class ClienteCreate(ClienteBase):
    pass


class ClienteUpdate(BaseModel):
    nome: str | None = None
    email: EmailStr | None = None
    telefone: str | None = None


class ClienteStatusUpdate(BaseModel):
    ativo: bool


class ClienteResponse(ClienteBase):
    id: int
    ativo: bool
    criado_em: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/cliente.ts
export interface Cliente {
  id: number
  nome: string
  email: string
  telefone: string | null
  ativo: boolean
  criado_em: string
}

export interface ClienteCreate {
  nome: string
  email: string
  telefone?: string | null
}

export interface ClienteStatusUpdate {
  ativo: boolean
}